tests/test_templater.py

```diff
@@ -291,3 +293,62 @@ def test_set_template_verbosity():
     outcome = ''
     # Check that only the correct messages were logged
     assert result == outcome
+
+def test_log_decorator():
+    """Unit test for checking application of the logger decorator"""
+
+    input_file = os.path.join(uwtools_file_base, "fixtures/nml.IN")
+    logfile = os.path.join(os.path.dirname(templater.__file__), "templater.log")
+
+    with tempfile.TemporaryDirectory(dir='.') as tmp_dir:
+        out_file = f'{tmp_dir}/test_render_from_yaml.nml'
+
+        outcome=\
+    """Finished setting up debug file logging in """ + logfile  + """
+Running script templater.py with args:
+----------------------------------------------------------------------
+----------------------------------------------------------------------
+        outfile: """ + out_file  + """
+ input_template: """ + input_file  + """
+    config_file: None
+   config_items: []
+        dry_run: False
+  values_needed: False
+        verbose: True
+          quiet: False
+----------------------------------------------------------------------
+----------------------------------------------------------------------
+Test calling templater logging object
+Only the templater logger would see this
+
+&salad
+  base = 'kale'
+  fruit = 'banana'
+  vegetable = 'tomato'
+  how_many = 22
+  dressing = 'balsamic'
+/
+"""
+
+        args = [
+             '-i', input_file,
+             '-o', out_file,
+             '-v'
+             ]
+
+        print("Test templater with logger decorator")
+        # Capture verbose stdout
+        outstring = io.StringIO()
+        with redirect_stdout(outstring):
+            templater.set_template(args)
+            print("Test calling templater logging object")
+            #Note: all instantiated loggers are stored in logging.root.manager.loggerDict
+            log = logging.getLogger('test_templater.uwtools.templater')
```

Comment:
Should `logging` here be the UW Logger... maybe like this:

```
log = Logger(name='test_templater.uwtools.templater').get_logger()
```

There's a chance that Logger needs to be modified to make sure that if a logger object is already created with the same name it just returns that object that exists....
In __init__ for UW Logger, add a debug statement for whether it matches an existing object or is creating a new one (whether its in `logging.root.manager.loggerDict`).

Reply:
Turns out there's an open pylint issue on this, just have to disable it